pkg/cli/tasks.go

```diff
@@ -0,0 +1,296 @@
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/spf13/cobra"
+)
+
+type Tasks struct {
+	root        *Obot
+	Quiet       bool   `usage:"Only print IDs of tasks" short:"q"`
+	Wide        bool   `usage:"Print more information" short:"w"`
+	Output      string `usage:"Output format (table, json, yaml)" short:"o" default:"table"`
+	ThreadID    string `usage:"Filter tasks by Thread ID" short:"t"`
+	AssistantID string `usage:"Filter tasks by Assistant ID" short:"a"`
+	ProjectID   string `usage:"Filter tasks by Project (Obot) ID" short:"p"`
+	NoRuns      bool   `usage:"Don't fetch run counts (faster)" short:"n"`
+	MaxTasks    int    `usage:"Maximum number of tasks to process per thread" default:"10"`
+	All         bool   `usage:"List all tasks (admin only)" short:"A"`
+}
+
+func (l *Tasks) Customize(cmd *cobra.Command) {
+	cmd.Use = "tasks [flags] [TASK_ID...]"
+	cmd.Aliases = []string{"task", "ts"}
+}
+
+func (l *Tasks) Run(cmd *cobra.Command, args []string) error {
+	debugPrint := func(format string, args ...interface{}) {
+		if l.root.Debug {
+			fmt.Fprintf(cmd.OutOrStdout(), format+"\n", args...)
+		}
+	}
+
+	ctx := cmd.Context()
+	start := time.Now()
+
+	// If ProjectID is provided but AssistantID is not, look up the assistant ID
+	if l.ProjectID != "" && l.AssistantID == "" {
+		debugPrint("Looking up assistant ID for project %s", l.ProjectID)
+		project, err := l.root.Client.GetProject(ctx, l.ProjectID)
+		if err != nil {
+			return fmt.Errorf("failed to get project %s: %w", l.ProjectID, err)
+		}
+		l.AssistantID = project.AssistantID
+		debugPrint("Found assistant ID %s for project %s", l.AssistantID, l.ProjectID)
+	}
+
+	var (
+		allTasks types.TaskList
+	)
+
+	// 1. First fetch projects to know what we have access to
+	debugPrint("Fetching projects...")
+	startProjects := time.Now()
+
+	projects, err := l.root.Client.ListProjects(ctx, apiclient.ListProjectsOptions{
+		All: l.All,
+	})
+	if err != nil {
+		debugPrint("Error listing projects: %v", err)
+		return fmt.Errorf("failed to list projects: %w", err)
+	}
+	debugPrint("Fetched %d projects in %.2f seconds", len(projects.Items), time.Since(startProjects).Seconds())
+
+	// Build lookup maps for quick access
+	projectMap := make(map[string]types.Project)
+	for _, project := range projects.Items {
+		projectMap[project.ID] = project
+	}
+
+	// If specific project is requested, only get tasks for that project
+	if l.ProjectID != "" && l.AssistantID != "" {
+		debugPrint("Using specific project %s with assistant %s", l.ProjectID, l.AssistantID)
+		tasks, err := l.root.Client.ListProjectTasks(ctx, l.AssistantID, l.ProjectID)
+		if err != nil {
+			return fmt.Errorf("failed to list tasks for project %s: %w", l.ProjectID, err)
+		}
+		allTasks = tasks
+	} else {
+		// Otherwise aggregate tasks from all accessible projects
+		taskChan := make(chan types.Task)
+		errChan := make(chan error, len(projects.Items))
+		doneChan := make(chan bool)
```

Comment:
The typical way to do this is to change this channel to the following and close it instead of passing a bool.
```suggestion
		doneChan := make(chan struct{})
```